website/utils/url_utils.ts

```diff
@@ -26,15 +30,21 @@ const SLICE_REGEX = /^[0-9.]*,[0-9.]*,[0-9.]*$/;
 const REGION_REGEX = /^([0-9.]*:[0-9.]*)(,[0-9.]*:[0-9.]*){2}$/;
 const HEX_COLOR_REGEX = /^[0-9a-fA-F]{6}$/;
 
-export type ViewerChannelSettingJson = {
+// Classes have default `undefined` values for all properties
+// to ensure the keys exist when the object is created with no arguments.
+// This allows us to iterate over the keys and turn them into a string list of allowed keys.
+/**
+ * The serialized form of a ViewerChannelSetting, as a dictionary object.
+ */
+export class ViewerChannelSettingParams {
   /** Color, as a 6-digit hex color.  */
-  col?: string;
+  col?: string = undefined;
```

Comment:
suggestion: make these all enum values so you can do `[urlParam.Color]: string = undefined` to make it more readable